protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/RecordMetadata.java

```diff
@@ -84,6 +86,13 @@ public void wrap(final DirectBuffer buffer, int offset, final int length) {
                         versionDecoder.patchVersion()))
             .orElse(VersionInfo.UNKNOWN);
 
+    final int decodedRecordVersion = decoder.recordVersion();
+    if (decodedRecordVersion > 0) {
+      recordVersion = decodedRecordVersion;
+    } else {
+      recordVersion = DEFAULT_RECORD_VERSION;
+    }
```

Comment:
❓ I'm a bit confused how this works. The default of the `recordVersion` is 1. So this means it is always filled for new records. For older records where we don't have a `recordVersion` what does this decoder return?

```java
    public int recordVersion()
    {
        if (parentMessage.actingVersion < 3)
        {
            return 65535;
        }

        return (buffer.getShort(offset + 30, java.nio.ByteOrder.LITTLE_ENDIAN) & 0xFFFF);
    }
```

It seems it can return `65535` which is higher than 0 as well, but I'm not sure what this if check means.

Reply:
I'm not sure about this, I've asked the team [on slack](https://camunda.slack.com/archives/CSQ2E3BT4/p1685971972989149). I think we can ignore it, as our UpdateTests should cover processing (and applying) of records written on old versions of Zeebe.

Reply:
So, 65535 is the nullValue. See https://www.fixtrading.org/standards/sbe-online/#integer-encoding

This means that @remcowesterhoud found a bug in my code 🎉

Reply:
Oops, it seems that using 65535 instead of 0 actually caused a bug 😅 See b981024.